app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/TogglePlaybackStateSwipeAction.java

```diff
@@ -39,6 +39,6 @@ public void performAction(FeedItem item, Fragment fragment, FeedItemFilter filte
 
     @Override
     public boolean willRemove(FeedItemFilter filter, FeedItem item) {
-        return filter.showUnplayed || filter.showPlayed || filter.showNew;
+        return filter.showPlayed || filter.showNew;
```

Comment:
I think you need to check based on the item here. Otherwise the items are no longer swiped out completely when filtering by "unplayed" and expecting the toggle action to remove from screen. So, when the item is new, do what you do now. Otherwise do what was done before.

Reply:
Thanks, I'll work on this over this weekend.